zvariant/src/file_path.rs

```diff
@@ -0,0 +1,138 @@
+use std::{borrow::Cow, path::{Path, PathBuf}};
+use serde::{de::{self, Visitor}, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::Type;
+
+/// A file name represented as a nul-terminated byte array.
+///
+/// While `zvariant::Type` and `serde::{Serialize, Deserialize}`, are implemented for [`Path`] and [`PathBuf`], unfortunately `serde` serializes them as UTF-8 strings. This is not the desired behavior in most cases since file paths are not guaranteed to contain only UTF-8 characters.
+/// To solve this problem, this type is provided which encodes the underlying file path as a null-terminated byte array. Encoding as byte array is also more efficient.
+///
+///
+/// # Exmples
+/// ```
+/// use zvariant::FilePath;
+/// use std::path::{Path, PathBuf};
+///
+/// let path = Path::new("/hello/world");
+/// let path_buf = PathBuf::from(path);
+///
+/// let p1 = FilePath::from(path);
+/// let p2 = FilePath::from(path_buf);
+/// let p3 = FilePath::from("/hello/world");
+///
+/// assert_eq!(p1, p2);
+/// assert_eq!(p2, p3);
+/// ```
```

Comment:
I added some tests for the (de)serialization process, but didn't change the examples other than adding some `Into`s, since I really think that the way the type is (de)serialized should not matter for users, as they should be able to just wrap their file paths in it and send/receive it.

Reply:
> I really think that the way the type is (de)serialized should not matter for users

As I wrote, that's the main use here. Users will not be using this a generic type for file paths, but OK, no biggie.